### PR TITLE
Fixes #56 about taking over bash aliases into xonsh.

### LINKED ISSUE #56: Take over bash aliases?
Custom bash aliases (eg. in ~/.bash_aliases) do not work after starting xonsh from bash.

Maybe documenting how to best port them to xonsh-aliases ist enough.

### PR DESCRIPTION
**Pull request by ivh**
Not sure if the split and stripping of 'alias ' and the single quotes is right, but it seems to work here.

I chose to add it to aliases.py instead of extending bash_env() in environ.py since it felt better that way.

BTW, in built_ins.py calling builtins.aliases.update instead of builtins.aliases._raw.update failed with NoneType iterable error.

### CODE REVIEW

File: xonsh/aliases.py [hunk 1]
```diff
@@ -55,6 +56,20 @@ def source_bash(args, stdin=None):
...
 
+def bash_aliases():
+    try:
+        s = subprocess.check_output(['bash', '-i'], input='alias',
+                                    stderr=subprocess.PIPE,
+                                    universal_newlines=True)
+    except subprocess.CalledProcessError:
+        s = ''
+    items = [line.split('=', 1) for line in s.splitlines() if '=' in line]
+    aliases = [(item[0][6:],shlex.split(item[1].strip('\''))) for item in items]
```
Comment (scopatz) [inline xonsh/aliases.py]: PEP8 Space after comma

File: xonsh/aliases.py [hunk 2]
```diff
@@ -55,6 +56,20 @@ def source_bash(args, stdin=None):
...
+    return dict(aliases)
+
+
+BASH_ALIASES = bash_aliases()
```
Comment (scopatz) [inline xonsh/aliases.py]: hmm, I think it would be cleaner to not have this here and instead import the `bash_aliases()` function in built_ins.

**Comment by scopatz** [comment]
This fails travis for some reason that seems unrelated.  Maybe try rebasing off of master?

**Comment by scopatz** [comment]
Thanks a ton for putting this in @ivh!  Just a few minor issues to address (ping when done.)

> BTW, in built_ins.py calling builtins.aliases.update instead of builtins.aliases._raw.update failed with NoneType iterable error.

This was because I forgot a `self` in this method.  It should work on master now (444bc33).

**Comment by ivh** [comment]
@scopatz Ok, I've done the changes you asked for in the same branch and did a pull --rebase from your master before commiting them.

(I stared at Aliases.update for a few seconds and thought "this shouldn't fail!" but did not see the missing self - typical.)

File: xonsh/aliases.py [hunk 3]
```diff
@@ -55,6 +56,21 @@ def source_bash(args, stdin=None):
...
+                                    universal_newlines=True)
+    except subprocess.CalledProcessError:
+        s = ''
+    items = [line.split('=', 1) for line in s.splitlines() if '=' in line]
+    aliases = [(item[0][6:], shlex.split(item[1].strip('\''))) \
+                for item in items]
+    return dict(aliases)
+
+
+BASH_ALIASES = bash_aliases()
```
Comment (scopatz) [inline xonsh/aliases.py]: This is probably what is causing the CI failure.

**Comment by ivh** [comment]
@scopatz sorry, forgot to remove that loine, fixed now.

**Comment by scopatz** [comment]
Thanks @ivh.  I think that this needs to be a little safer than it is now. For example, I have the following Bash alias:

```bash
alias git-commit-count='git log --pretty=format:'\'''\'' | wc -l'
```

This causes the following error in the PR:

```console
Traceback (most recent call last):
  File "/home/scopatz/.local/bin/xonsh", line 2, in <module>
    from xonsh.main import main
  File "/home/scopatz/.local/lib/python3.4/site-packages/xonsh/main.py", line 8, in <module>
    from xonsh.shell import Shell
  File "/home/scopatz/.local/lib/python3.4/site-packages/xonsh/shell.py", line 10, in <module>
    from xonsh.execer import Execer
  File "/home/scopatz/.local/lib/python3.4/site-packages/xonsh/execer.py", line 13, in <module>
    from xonsh.built_ins import load_builtins, unload_builtins
  File "/home/scopatz/.local/lib/python3.4/site-packages/xonsh/built_ins.py", line 19, in <module>
    from xonsh.aliases import DEFAULT_ALIASES, bash_aliases
  File "/home/scopatz/.local/lib/python3.4/site-packages/xonsh/aliases.py", line 75, in <module>
    BASH_ALIASES = bash_aliases()
  File "/home/scopatz/.local/lib/python3.4/site-packages/xonsh/aliases.py", line 71, in bash_aliases
    print(shlex.split(item[1].strip('\'')))
  File "/home/scopatz/miniconda3/lib/python3.4/shlex.py", line 276, in split
    return list(lex)
  File "/home/scopatz/miniconda3/lib/python3.4/shlex.py", line 266, in __next__
    token = self.get_token()
  File "/home/scopatz/miniconda3/lib/python3.4/shlex.py", line 93, in get_token
    raw = self.read_token()
  File "/home/scopatz/miniconda3/lib/python3.4/shlex.py", line 169, in read_token
    raise ValueError("No closing quotation")
ValueError: No closing quotation
```

Since this function gets called on startup, this prevents xonsh from even opening. We should definitely skip - and possibly warn - if an aliase cannot be parsed.

**Comment by ivh** [comment]
That is a strangely looking alias - did GitHub mess up the string you pasted? (It has unclosed quotes)

But yes, I can have another look at the parsing and skip/warn when it fails.

**Comment by scopatz** [comment]
Sadly that is a valid bash alias. I have been using it for years. I don't think that it is the kind of thing that we should support in xonsh though :)

**Comment by scopatz** [comment]
Also, github did not mess up. I just double checked - that is exactly what is in bash.

**Comment by ivh** [comment]
I just added the warning and skipping of an alias when it does not compute.

However, I could not easily get your example to work by playing around with shlex options like posix or escapequote.

**Comment by scopatz** [comment]
Thanks for this @ivh!  I did a minor amount of cleanup and then merged. FYI. Super happy to see this in xonsh.

**Comment by ivh** [comment]
Glad it made it in!

**Pull request merged**